lib/screens/connectPage/connect.dart

```diff
@@ -0,0 +1,226 @@
+import 'dart:async';
+import 'package:feetback/screens/homePage/home.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bluetooth_serial/flutter_bluetooth_serial.dart';
+import 'package:flutter_spinkit/flutter_spinkit.dart';
+import 'widgets/BluetoothDeviceListEntry.dart';
+import '../../services/bluetooth_service.dart';
+
+class DiscoveryPage extends StatefulWidget {
+  /// If true, discovery starts on page start, otherwise user must press action button.
+  final bool start;
+
+  const DiscoveryPage({this.start = true});
+
+  @override
+  _DiscoveryPage createState() => new _DiscoveryPage();
+}
+
+class _DiscoveryPage extends State<DiscoveryPage> {
+  StreamSubscription<BluetoothDiscoveryResult> _streamSubscription;
+  List<BluetoothDiscoveryResult> results = List<BluetoothDiscoveryResult>();
+  bool isDiscovering;
+  _DiscoveryPage();
+
+  @override
+  void initState() {
+    super.initState();
+    _startApp();
+  }
+
+  Future<void> _startApp() async {
+    await Session.shared.checkBluetoothAdapter(this);
```

Comment:
The name 'Session' is not clear. If this belongs to the bluetooth service please make it clear by using a BluetoothService instance.
What is the purpose of shared? Can't 'checkBluetoothAdapter' be a direct method of BluetoothService?